Assets/Scripts/Environment/Hangar/BatterySpawnPositionManager.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BatterySpawnPositionManager : MonoBehaviour
+{
+    [SerializeField] private GameObject[] spawnPositions;
+
+    [SerializeField] private GameObject[] batteries;
+
+    public void Awake()
+    {
+        // As the game starts, spawn the batteries at their initial position
+        for (int i = 0; i < batteries.Length; i++)
+        {
+            batteries[i].transform.position = spawnPositions[i].transform.position;
+        }
+    }
+
+    // Respawn the battery when the player tries to take it out of the hangar
+    public void RespawnBattery(Transform objectToMove, Collider boundaries)
+    {
+        Collider[] colliders;
+
+        // We loop through the spawn positions until we find one that is free
+        for (int i = 0; i < spawnPositions.Length; i++)
+        {
+            // Check if there is already a gameobject within the space where we want to spawn the battery
+            colliders = Physics.OverlapBox(spawnPositions[i].transform.position, boundaries.bounds.extents, Quaternion.identity);
+
+            // If Physics.OverlapBox() returns nothing, it means that the space is free
+            if(colliders.Length == 0)
+            {
+                objectToMove.transform.position = spawnPositions[i].transform.position;
```

Comment:
Will the battery's parent still be the player when the battery gets respawned? As the battery gets parented to the player on pickup. This could have side effects if so. 

Although if we use the joint solution this won't be an issue as there's no parent setting involved.

Reply:
Ignore this comment above. I just saw that `DropBattery` gets called before `RespawnBattery`, which will reparent it.

Reply:
got it